package-res/resources/web/pentaho/type/list.js

```diff
@@ -256,70 +310,75 @@ define([
        * @return {pentaho.type.List} The added list element.
        */
       add: function(fragment) {
-        this._set(fragment, /*add:*/true, /*update:*/true, /*remove:*/false);
+        return this._set(fragment, /*add:*/true, /*update:*/true, /*remove:*/false, /*move:*/false);
       },
 
       /**
-       * Inserts or updates one or more elements, starting at the given index.
+       * Inserts and/or updates one or more elements, starting at the given index.
        *
        * If `index` is negative,
        * it means the position at that many elements from the end (`index' = count - index`).
        *
        * When the index is greater than or equal to the length of the list,
        * the element or elements are appended to the list.
        *
-       * @param {any|any[]} fragment Element or elements to add.
+       * @param {any|Array} fragment Element or elements to add.
        * @param {number} index The index at which to start inserting new elements.
```

Comment:
Suggest changing to:

index - The index at which to start inserting new elements.